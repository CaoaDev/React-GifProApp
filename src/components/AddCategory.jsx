import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( { setAnime } ) => {
    const [inputValue, setImputValue] = useState( '' )
    const handleInputChange = ( e ) => {
        setImputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        console.log( 'handleSubmit', inputValue );
        if ( inputValue.trim().length >= 1 ) {
            setAnime ( anime => [ inputValue, ...anime ] );
            setImputValue ('');
        }
    }
    // evento para limpiar imput por onclick
    // const clean = (e) => {
    //     e.preventDefault();
    //     setImputValue('')
    // }
    // funcion para agregar una lista en el setAnimes
    // const handleAdd = () => {
    //     setAnimes( animes => [...animes, 'HunterxHunter', 'Madoka Magica', 'Kimetsu'] );
    // }

    return (
        <form onSubmit={ handleSubmit }>
            <p> { inputValue } </p>    
            <h1>Agregando: { inputValue }</h1>
             < input
             type= 'text'
             value={ inputValue }
             onChange={ handleInputChange }
            //  onClick={ clean }
             placeholder='Nuevo Anime... '
             />
             </form>
             )
            }
            
            AddCategory.propTypes = {
                setAnime: PropTypes.func.isRequired
            }