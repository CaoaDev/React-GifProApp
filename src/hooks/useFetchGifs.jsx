import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs =( anime ) => {
    const [first, setFirst] = useState({
        images: [],
        loading: true
    });

    useEffect( () => {
        getGifs ( anime )
        .then ( imgs => {
                setFirst ({
                    images: imgs,
                    loading: false
                });
        })
    },[ anime ]);
    
    return first;
}