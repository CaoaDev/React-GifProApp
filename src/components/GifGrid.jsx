import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
// import { getgif } from '../helpers/getGifs';

export const GifGrid = ( { anime } ) => {
  const { images, loading } = useFetchGifs( anime );

  // const [images, setImages] = useState([]);
  // useEffect( () => {
  //   getgif( anime )
  //   .then( setImages );
  // }, [ anime ]);

  return (
    <div>
      <h3 className= 'animate__bounceOutDown'>{ anime } </h3> 
      { loading && <p className= 'animate__fadeIn'>Loading</p>}
      <div className='card-grid'>
        {
          images.length > 0
            ?
              images.map ( ( img )=>(
                <GifGridItem 
                  key = { img.id }
                  { ...img }
                />
              ))
            :
              <p>no se encontro informacion de { anime }</p>
        }
      </div>
    </div>
  )
};

GifGrid.prototype = {
  category: PropTypes.string.isRequired
}