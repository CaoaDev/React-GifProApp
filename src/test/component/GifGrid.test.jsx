import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('pruebas en el <GifGrid />', () => {

    const anime= 'one punch';

    test('Hacer toMatchSnapshot correcatamente del GifGrid', () => {
        
        useFetchGifs.mockReturnValue({
            images: [],
            loading: true
        });

        const wrapper= shallow( <GifGrid anime={ anime }  /> );

        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe mostrar los valores cuando se cargan las imagenes useFetchGifs', () => {
        
        const gif= [{
            id: 'ABC',
            url: 'https://local/host.jpg',
            title: 'titulo'
        },
        {
            id: '123',
            url: 'https://local/host.jpg',
            title: 'titulo'
        }]

        useFetchGifs.mockReturnValue({
            images: gif,
            loading: false
        });

        const wrapper= shallow( <GifGrid anime={ anime }  /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gif.length )
    });
    
});