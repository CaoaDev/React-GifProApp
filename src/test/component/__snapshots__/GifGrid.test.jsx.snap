// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <GifGrid /> Debe mostrar los valores cuando se cargan las imagenes useFetchGifs 1`] = `
<div>
  <h3
    className="animate__bounceOutDown"
  >
    one punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="titulo"
      url="https://local/host.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="titulo"
      url="https://local/host.jpg"
    />
  </div>
</div>
`;

exports[`pruebas en el <GifGrid /> Hacer toMatchSnapshot correcatamente del GifGrid 1`] = `
<div>
  <h3
    className="animate__bounceOutDown"
  >
    one punch
     
  </h3>
  <p
    className="animate__fadeIn"
  >
    Loading
  </p>
  <div
    className="card-grid"
  >
    <p>
      no se encontro informacion de 
      one punch
    </p>
  </div>
</div>
`;
